src/Middleware/DistributedCacheInvalidation.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/*
+ * This file is part of fof/redis.
+ *
+ * Copyright (c) Bokt.
+ * Copyright (c) Blomstra Ltd.
+ * Copyright (c) FriendsOfFlarum
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace FoF\Redis\Middleware;
+
+use Flarum\Foundation\Paths;
+use Flarum\Locale\LocaleManager;
+use Illuminate\Contracts\Redis\Factory as Redis;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * Middleware to handle distributed cache invalidation across multiple instances.
+ *
+ * When cache is cleared on one instance, this middleware detects the change
+ * and invalidates local file caches on other instances.
+ */
+class DistributedCacheInvalidation implements MiddlewareInterface
+{
+    /**
+     * Timestamp of last check (per PHP-FPM worker).
+     *
+     * @var int
+     */
+    private static $lastCheck = 0;
+
+    /**
+     * How often to check Redis for cache version changes (seconds).
+     *
+     * @var int
+     */
+    private const DEFAULT_CHECK_INTERVAL = 5;
```

Comment:
I, personally, find this a bit too often, as only clearing the cache regularly takes over 5 seconds.